src/module.c

```diff
@@ -2486,6 +2506,342 @@ RedisModuleString *RM_ListPop(RedisModuleKey *key, int where) {
+    if (listTypeNext(li, &key->u.list.entry)) {
+        key->u.list.hascurrent = 1;
+        if (elem != NULL) {
+            robj *encoded = listTypeGet(&key->u.list.entry);
+            *elem = getDecodedObject(encoded);
+            decrRefCount(encoded);
+            serverAssert((*elem)->type == OBJ_STRING);
+            autoMemoryAdd(key->ctx, REDISMODULE_AM_STRING, *elem);
+        }
+        return REDISMODULE_OK;
+    } else {
+        key->u.list.hascurrent = 0;
+        errno = ENOENT;
+        return REDISMODULE_ERR;
+    }
+}
+
+/* Deletes the list element at the current position in a list iterator started
+ * on a key.
+ *
+ * Returns REDISMODULE_OK on success. Returns REDISMODULE_ERR on failure and
+ * sets `errno` as follows:
+ *
+ * - EINVAL if key is NULL.
+ * - ENOTSUP if the key is not a list.
+ * - EBADF is no iterator has been started.
+ * - ENOENT if there is no current element.
+ */
+int RM_ListIteratorDelete(RedisModuleKey *key) {
+    if (!key) {
+        errno = EINVAL;
+        return REDISMODULE_ERR;
+    } else if (!key->value || key->value->type != OBJ_LIST) {
+        errno = ENOTSUP;
+        return REDISMODULE_ERR;
+    } else if (!(key->mode & REDISMODULE_READ) || !key->iter) {
+        errno = EBADF;
+        return REDISMODULE_ERR;
+    } else if (!key->u.list.hascurrent) {
+        /* TODO? Reseek automatically to allow Delete() without Next() */
+        errno = ENOENT;
+        return REDISMODULE_ERR;
+    }
+
+    listTypeIterator *li = key->iter;
+    listTypeDelete(li, &key->u.list.entry);
+    /* if (&key->u.list.entry.entry.value) */
+    /*     decrRefCount((robj *)&key->u.list.entry.entry.value); */
+    key->u.list.hascurrent = 0;
+    return REDISMODULE_OK;
+}
+
+/* Inserts an element before (where = REDISMODULE_LIST_BEFORE) or after (where =
+ * REDISMODULE_LIST_AFTER) the current element in a list during iteration. The
+ * current element is the element returned by the last call to
+ * RM_ListIteratorNext().
+ *
+ * Returns REDISMODULE_OK on success and on failure REDISMODULE_ERR is returned
+ * and errno is set as follows:
+ *
+ * - EINVAL if key is NULL.
+ * - ENOTSUP if the key is not a list.
+ * - EBADF is no iterator has been started.
+ * - ENOENT if there is no current element.
+ */
+int RM_ListIteratorInsert(RedisModuleKey *key, int where, RedisModuleString *value) {
+    if (!key) {
+        errno = EINVAL;
+        return REDISMODULE_ERR;
+    } else if (!key->value || key->value->type != OBJ_LIST) {
+        errno = ENOTSUP;
+        return REDISMODULE_ERR;
+    } else if (!(key->mode & REDISMODULE_READ) || !key->iter) {
+        errno = EBADF;
+        return REDISMODULE_ERR;
+    } else if (!key->u.list.hascurrent) {
+        errno = ENOENT;
+        return REDISMODULE_ERR;
+    }
+
+    listTypeIterator *li = key->iter;
+    where = (where == REDISMODULE_LIST_BEFORE) ? LIST_HEAD : LIST_TAIL;
+    listTypeInsert(&key->u.list.entry, value, where);
+    key->u.list.reseek = 1;
```

Comment:
Mhm... listTypeIterator can handle deletes but not inserts. I think it's the quicklists that must be rewritten. So, it may be possible but it's not trivial.